### PR TITLE
Qt gui xdg theme support

### LINKED ISSUE #720: Support for xdg icon themes in qt-gui
So i looked into getting xdg icon theme support into the qt-gui and would like to hear the opinion of the rest of the team in regard of mapping icons that are not straight mappable to the [KDE](https://techbase.kde.org/Projects/Oxygen/namingSpec)/[GNOME](https://developer.gnome.org/icon-naming-spec/)/[XDG](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html) icon theme specifications.

Getting xdg icon theme support into the qt-gui allows people with impaired vision to use a icon set that works better for them. e.g. HighContrast Icon set

The current approach would look first if a theme icon is available, if not it would fallback to the icon shipped with the application.

fallback Icon | icon for | proposed icon name | Breeze | Oxygen Icon | Adwaita Icon | HC Icon | notes
-----------------|------------|-----------------|-------------------|-------------------|----------------|---------|-----
document-close | deleting key/array | edit-delete | edit-delete | edit-delete | edit-delete | edit-delete | original icon document-close is kde specific, edit-delete is part of xdg spec
|| _alternative_ | user-trash | user-trash | user-trash | user-trash | user-trash | 
|| _alternative_ | edit-clear | edit-clear | edit-clear | edit-clear | edit-clear
import | Import Configuration | document-open | document-open | document-open | ![document-open] (https://cloud.githubusercontent.com/assets/1131371/15274093/6881679c-1aab-11e6-8d61-a3338e013da9.png) | document-open | original icon document-import kde specific, document-open is part of xdg spec
export | Export Configuration | document-save | document-save | document-save | document-save | document-save | original icon document-export is kde specific, document-save is part of xdg spec
|| _alternative_ | document-save-as | document-save-as | document-save-as | document-save-as | document-save-as | maybe save-as is a better representation for export then save? 
mount | Mount Backend | list-add | list-add | list-add | list-add | list-add | This is the best i could find in the xdg spec for mount and unmount, they should be clear as long as they are shown in context.
unmount | Unmount Backend | list-remove | list-remove | list-remove | list-remove | list-remove
new-array | New Array | view-grid | view-grid | view-grid | | view-grid | Grid as abstract visualisation of array. Is this to much of a stretch?
color | Color Settings | preferences-desktop-color | preferences-desktop-color | preferences-desktop-color ||| kde specific
|| _alternative_ | preferences-color ||| preferences-color | preferences-color | gnome specific. Maybe find a solution to give the image provider more then one icon name
|| _alternative_ | applications-system | applications-system | applications-system | applications-system | preferences-system
edit-rename | edit key/array | edit-rename | edit-rename | edit-rename ||| kde specific
|| _alternative_ | accessories-text-editor | accessories-text-editor | accessories-text-editor | accessories-text-editor | accessories-text-editor
|| _alternative_ | document-properties | document-properties | document-properties | document-properties | document-properties
whats_this | contextual help | help-contextual | help-contextual | help-contextual | help-contents |  help-contents | help-contextual is kde specific it automaticly falls back to default help icon on gnome

### PR DESCRIPTION
**Pull request by sirblackheart**
This is a rework of #751 divided into several smaller commits as requested. I also tried to fix all the concerns brought up in the previous pull request. As this is a rather big pull request, I hope we get enough people to test this, so we can make sure no regressions are introduced trough it.

I tried to describe all changes and the reasoning behind them in the commit messages. If i missed any feel free to comment on it.

And as a reminder, this change introduces a new dependency for the qt-gui tool, the qt5 `svg` module.

close #720

### CODE REVIEW

**Comment by markus2330** [comment]
Now everything looks perfect. Thank you!

@0003088: Do you want to take a further look and try it out? Can you look at the QML?

Btw1: Actually I said you do not need to rework the commits, I only mentioned the issue for further PRs.
Btw2: The first line of the commit msg ideally should already tell something about a commit, usually we use `component: short change description`, where component would be qt-gui in this case, e.g.: `qt-gui: introduce helper functions`. Again: please do not rework it for this PR. It really is *not* important and your commit msgs already are great.

**Comment by markus2330** [comment]
Somehow every commit changed, its hard to tell now what exactly was changed since I looked at it last time?

**Comment by sirblackheart** [comment]
I did rebase sorry, the cmake install instruction had still the old desktop file name.
There are two new commits. Ones is for the symbolic icon. GNOME Shell uses them. I had to slightly alter the elektra logo so it still looks for that size (16 x 16 pixel for symbolic icons). The other adds appstream data. That is metadata used by `gnome-software-center` in addition to the `desktop` file entries.

**Comment by markus2330** [comment]
Ok, looks both good to me. @0003088 seems to be off for the weekend. @sirblackheart is it okay for you to wait?

**Comment by sirblackheart** [comment]
@markus2330 sure, i rather have this looked over by @0003088 then introduce any regression

**Comment by sirblackheart** [comment]
I am still open to drop the Qt in the desktop name altogether and use just Elektra Editor (elektra-editor) for it as public name, id and process-name, as right now it is the only native GUI available for Elektra. In addition before this commits, the GUI referred to itself as Elektra Editor.

Internally we could still distinguish between `kdb editor` as core tool and `Elektra Editor` and `qt-gui` as the Qt GUI.

This is not a suggestion to change the tool name, that probably should stay the same, as it make sense when someone starts it with `kdb qt-gui`.

**Comment by markus2330** [comment]
I think we should keep Qt as part of its name. Elektra should not be positioned towards a specific desktop dnvironment or toolkit. It is a technology to be used in any desktop environment or toolkit.

As you know `kdb editor` is someting entirely different: it uses any (text) editor to edit configuration. People were unhappy with the name `kdb edit` (see #388). We picked `kdb editor` because it actually spawns a tool called `editor` (or `sensible-editor` if available).
New name suggestions can be done in #265.

File: src/tools/qt-gui/src/guisettings.hpp [hunk 1]
```diff
@@ -124,6 +137,11 @@ public slots:
 	 */
 	void nodeWithoutKeyColorChanged ();
 
+	/**
+	 * @brief This signal is emitted if the direction if the system icon theme should be used or not.
```
Comment (0003088) [inline src/tools/qt-gui/src/guisettings.hpp]: This is a strange sentence.

**Comment by 0003088** [comment]
I hope I have not delayed the pull request due to my absence. The changes look fine to me (and are a nice improvement to the gui)

File: src/tools/qt-gui/src/guisettings.hpp (hunk 1 above)
Comment (sirblackheart) [inline src/tools/qt-gui/src/guisettings.hpp]: Thank you for noticing. Too much copy paste at work.

**Comment by markus2330** [comment]
Thank you both then! @sirblackheart well done, is a nice improvement!

**Pull request merged**